aries-site/src/layouts/content/InPageNavigation.js

```diff
@@ -0,0 +1,126 @@
+import { useState, useRef, useEffect } from 'react';
+import Link from 'next/link';
+import PropTypes from 'prop-types';
+import { Box, Button, Nav, Text } from 'grommet';
+import { siteContents } from '../../data/search/contentForSearch';
+import { nameToSlug } from '../../utils';
+
+const useActiveHeadingId = (headings, options) => {
+  const [activeHeadingId, setActiveHeadingId] = useState();
+  const observer = useRef();
+  // useEffect finds which page heading is on screen as the user scrolls
+  useEffect(() => {
+    // if at top of page, do not show active heading on ToC
+    window.onscroll = () => {
+      if (window.pageYOffset === 0) {
+        setActiveHeadingId(null);
+      } else {
+        // find page active page heading
+        const elements = headings.map(heading =>
+          document.getElementById(nameToSlug(heading[1])),
+        );
+        observer.current?.disconnect();
+        observer.current = new IntersectionObserver(entries => {
+          entries.forEach(entry => {
+            if (entry?.isIntersecting) {
+              setActiveHeadingId(entry.target.id);
+            }
+          });
+        }, options);
+        elements.forEach(el => {
+          if (el) {
+            observer.current?.observe(el);
+          }
+        });
+      }
+      return () => observer.current?.disconnect();
+    };
+  }, [headings, options]);
+  return activeHeadingId;
+};
+
+export const InPageNavigation = ({ title }) => {
+  const match = siteContents.find(
+    item => item.name.toLowerCase() === title.toLowerCase(),
+  );
+
+  const regexp = new RegExp(/#{1,} (...+?) ?~{2}/, 'g');
+  const headings = match && [...match.content.matchAll(regexp)];
+
+  // top and bottom margin values to calculate intersection window
+  const topMargin = '50';
+  const bottomMargin = '-95';
+  const activeId = useActiveHeadingId(headings, {
+    rootMargin: `${topMargin}% 0% ${bottomMargin}% 0%`,
+  });
+
+  return headings.length > 0 ? (
+    <Box
+      pad={{ horizontal: 'xsmall' }} // pad for keyboard focus
+      style={{
+        height: 'calc(100vh - 50px)',
+        overflowY: 'auto',
+        position: 'sticky',
+        marginTop: '50px',
+        flexShrink: 0,
+        top: '30px',
+      }}
+      width="small"
+      flex={false}
+    >
+      <Box pad="small" a11yTitle="Table of Contents Heading">
```

Comment:
This a11yTitle shouldn't be on this Box. The screenreader will naturally read the text. I think what we want is an `a11yTitle` on the `nav` that reads "Jump to section"

Reply:
If we do that, then the screenreader reads the following: 
<img width="724" alt="Screen Shot 2022-08-02 at 11 00 39 AM" src="https://user-images.githubusercontent.com/12522275/182442829-4b48d6ff-409b-4e46-aebb-03b662e18c3e.png">

Reply:
I also added a Jump to section {headingTitle} in the boxes below. It's probably a little redundant, but this way the screen reader actually indicates that it is a jump to section link. Is this ok?

Reply:
I don't think we need to add to each button necessarily if we have it once on the overall nav. Here's the MDN best practices and example: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/navigation_role#associated_wai-aria_roles_states_and_properties